### ISSUE TITLE
interface was invalid memory address or nil pointer dereference when type check

### ISSUE THREAD

**Issue by soulbalz** [traefik/yaegi#1015]
The following program `sample.go` triggers a panic:

```go
package main

import (
	"encoding/json"
	"fmt"
	"net/url"
)

func main() {
	body := []byte(`{
		"BODY_1": "VALUE_1",
		"BODY_2": "VALUE_2",
		"BODY_3": null,
		"BODY_4": {
			"BODY_1": "VALUE_1",
			"BODY_2": "VALUE_2",
			"BODY_3": null
		},
		"BODY_5": [
			"VALUE_1",
			"VALUE_2",
			"VALUE_3"
		]
	}`)

	values := url.Values{}

	var rawData map[string]interface{}
	err := json.Unmarshal(body, &rawData)
	if err != nil {
		fmt.Println("can't parese body")
	}

	for key, val := range rawData {
		switch val.(type) {
		case string, bool, float64:
			values.Add(key, fmt.Sprint(val))
		case nil:
			values.Add(key, "")
		case map[string]interface{}, []interface{}:
			jsonVal, err := json.Marshal(val)
			if err != nil {
				fmt.Println("can't encode json")
			}
			values.Add(key, string(jsonVal))
		}
	}
	fmt.Println(values.Get("BODY_1"))
	fmt.Println(values.Get("BODY_2"))
	fmt.Println(values.Get("BODY_3"))
	fmt.Println(values.Get("BODY_4"))
	fmt.Println(values.Get("BODY_5"))
}

```

Expected result:
```console
$ go run ./sample.go
VALUE_1
VALUE_2

{"BODY_1":"VALUE_1","BODY_2":"VALUE_2","BODY_3":null}
["VALUE_1","VALUE_2","VALUE_3"]
```

Got:
```console
$ yaegi ./sample.go
run: runtime error: invalid memory address or nil pointer dereference
goroutine 1 [running]:
runtime/debug.Stack(0x1, 0xc0002d8800, 0x40)
	/usr/local/Cellar/go/1.15.6/libexec/src/runtime/debug/stack.go:24 +0x9f
github.com/traefik/yaegi/interp.(*Interpreter).eval.func1(0xc0000c1c70)
	/Users/olaf/go/pkg/mod/github.com/traefik/yaegi@v0.9.11/interp/interp.go:485 +0xc5
panic(0x1a0ade0, 0x2126d50)
	/usr/local/Cellar/go/1.15.6/libexec/src/runtime/panic.go:969 +0x1b9
github.com/traefik/yaegi/interp.runCfg.func1(0xc000356210, 0xc000117400, 0xc0000c1a60)
	/Users/olaf/go/pkg/mod/github.com/traefik/yaegi@v0.9.11/interp/run.go:184 +0x247
panic(0x1a0ade0, 0x2126d50)
	/usr/local/Cellar/go/1.15.6/libexec/src/runtime/panic.go:969 +0x1b9
github.com/traefik/yaegi/interp.(*itype).id(0x0, 0xc00047e4e0, 0x199)
	/Users/olaf/go/pkg/mod/github.com/traefik/yaegi@v0.9.11/interp/type.go:1082 +0x3a
github.com/traefik/yaegi/interp._case.func5(0xc000356210, 0xc00007f5c0)
	/Users/olaf/go/pkg/mod/github.com/traefik/yaegi@v0.9.11/interp/run.go:2644 +0xfc
github.com/traefik/yaegi/interp.runCfg(0xc000117400, 0xc000356210)
	/Users/olaf/go/pkg/mod/github.com/traefik/yaegi@v0.9.11/interp/run.go:190 +0x87
github.com/traefik/yaegi/interp.(*Interpreter).run(0xc0000321c0, 0xc000116c00, 0xc000356000)
	/Users/olaf/go/pkg/mod/github.com/traefik/yaegi@v0.9.11/interp/run.go:121 +0x2b0
github.com/traefik/yaegi/interp.(*Interpreter).eval(0xc0000321c0, 0xc00045a400, 0x3d8, 0x7ffeefbffb9e, 0x9, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/Users/olaf/go/pkg/mod/github.com/traefik/yaegi@v0.9.11/interp/interp.go:577 +0x809
github.com/traefik/yaegi/interp.(*Interpreter).EvalPath(0xc0000321c0, 0x7ffeefbffb9e, 0x9, 0xc00045a000, 0x3d8, 0x0, 0x0, 0xc000110ea0)
	/Users/olaf/go/pkg/mod/github.com/traefik/yaegi@v0.9.11/interp/interp.go:399 +0xfa
main.runFile(0xc0000321c0, 0x7ffeefbffb9e, 0x9, 0xc0000403c0, 0x1)
	/Users/olaf/go/pkg/mod/github.com/traefik/yaegi@v0.9.11/cmd/yaegi/run.go:123 +0xae
main.run(0xc000190030, 0x1, 0x1, 0x13, 0x13)
	/Users/olaf/go/pkg/mod/github.com/traefik/yaegi@v0.9.11/cmd/yaegi/run.go:89 +0x7e5
main.main()
	/Users/olaf/go/pkg/mod/github.com/traefik/yaegi@v0.9.11/cmd/yaegi/yaegi.go:144 +0x415
```

**Comment by mpl** [comment]
This should be fixed as part of the ongoing work in #1017

**Comment by soulbalz** [comment]
> This should be fixed as part of the ongoing work in #1017

Thanks for your contribute on my issue but i was check on your build it's was fail on test
